airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -1027,10 +1032,12 @@ def var_print(var_name):
 
     if project_id is None:
         if var_name is not None:
-            logging.info(
-                'project not included in {var}: '
-                '{input}; using project "{project}"'.format(
-                    var=var_name, input=table_input, project=default_project_id))
+            log = LoggingMixin().logger
```

Comment:
There is no GoogleBigQueryBaseCursor in scope that can be used, correct me if I'm wrong.

Reply:
I was thrown because the function is being called a method, so I assumed it was part of BigQueryHook but isn't. LGTM.

Reply:
Sorry to bring this up _after_ the PR has been merged, but I think this pattern is less than ideal: using this means that all the cases outside of a class now have a log source of "airflow.utils.log.LoggingMixin.LoggingMixin":

```
In [7]: from airflow.utils.log.LoggingMixin import LoggingMixin

In [8]: l = LoggingMixin().logger

In [9]: l.name
Out[9]: 'airflow.utils.log.LoggingMixin.LoggingMixin'
```

Compare this with:

```
In [10]: from airflow.contrib.hooks.bigquery_hook import BigQueryHook

In [11]: BigQueryHook().logger.name
Out[11]: 'airflow.contrib.hooks.bigquery_hook.BigQueryHook'
```

That advantage of the latter is that different classes can be set at different level log. In a case like this where we need to log from outside a class we loose that ability.

That said what we do is probably not worse in many cases where we were just using the root logger, this just happens to be one of the few cases that was using a differently named logger. So ideally in cases like this we'd somehow pass `__module__` in, rather than getting the module from the class.

An example of how this is different:

```
In [2]: from airflow.contrib.hooks.bigquery_hook import BigQueryHook

In [3]: from airflow.utils.log.LoggingMixin import LoggingMixin

In [4]: lm = LoggingMixin()

In [5]: h = BigQueryHook()

In [6]: h.logger.info("hook")
[2017-09-13 09:16:14,422] {<ipython-input-6-d9ef8875e3af>:1} INFO - hook

In [7]: lm.logger.info("mixin")
[2017-09-13 09:16:22,798] {<ipython-input-7-f4ed3f594263>:1} INFO - mixin

In [8]: import logging

In [9]: logging.getLogger('airflow.contrib').setLevel(logging.ERROR)

In [10]: lm.logger.info("mixin")
[2017-09-13 09:17:58,043] {<ipython-input-10-f4ed3f594263>:1} INFO - mixin

In [11]: h.logger.info("hook")

In [12]: # Note how there was no log produced

In [13]: h.logger.error("but error still is")
[2017-09-13 09:19:32,870] {<ipython-input-13-0b7830c22006>:1} ERROR - but error still is
```

Reply:
(I think on balance this change is still massively better, just wanted to point this out. If I have time I'll see about a PR to fix this behaviour.)

Reply:
Good point @ashb . I think this should be addressed with a better way of getting the logger. LoggingMixin().logger seems a bit off anyway.

Reply:
Very valid issue. Thank you for pointing out. I will push a fix soon:
https://issues.apache.org/jira/browse/AIRFLOW-1605